### PR TITLE
Fixed issue with check_slice_indices and numpy ints

### PR DESCRIPTION
**Pull request by richardjgowers**
Made base.reader.check_slice_indices public (was _check_slice_indices)

Closes Issue #604

refreshed some coordinates.base docs to numpy style

Various pep8 fixes

### CODE REVIEW

**Comment by dotsdl** [comment]
Looks good to me. Ready to merge?

**Comment by kain88-de** [comment]
Skimming over it I've seen you touched the doc strings to make the numpy style conform. Very good intentions but right now they aren't. All parameters go under the Parameter heading for example.

File: package/MDAnalysis/coordinates/base.py [hunk 1]
```diff
@@ -152,22 +152,24 @@ class Timestep(object):
...
+          Whether this Timestep has position information [``True``]
+        *velocities*
+          Whether this Timestep has velocity information [``False``]
+        *forces*
+          Whether this Timestep has force information [``False``]
+        *dt*
+          The time difference between frames (ps).  If :attr:`time`
+          is set, then `dt` will be ignored.
+        *time_offset*
+          The starting time from which to calculate time (ps)
```
Comment (kain88-de) [inline package/MDAnalysis/coordinates/base.py]: [Numpy style doc-strings](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt) work a little bit different. All function parameters are under the `Parameters` heading. For keyword arguments you are supposed to add `optional` behind the type. So the correct numpy formating for the parameter list is.
 
```
Parameters
----------
n_atoms: int
  The total number of atoms this Timestep describes
positions: bool, optional
  Whether this Timestep has position information [``True``]
velocities: bool, optional
  Whether this Timestep has velocity information [``False``]
forces: bool, optional
  Whether this Timestep has force information [``False``]
dt: float, optional
  The time difference between frames (ps).  If :attr:`time`
  is set, then `dt` will be ignored.
time_offset: float, optional
  The starting time from which to calculate time (ps)
```

File: package/MDAnalysis/coordinates/base.py [hunk 2]
```diff
@@ -651,9 +659,12 @@ def volume(self):
...
 
-        :Returns:
-           A (3, 3) numpy.ndarray of unit cell vectors
+        Returns
+        -------
+        A (3, 3) numpy.ndarray of unit cell vectors
 
+        Examples
+        --------
         For example::
```
Comment (kain88-de) [inline package/MDAnalysis/coordinates/base.py]: you can remove this line when you have the example heading above.

File: package/MDAnalysis/coordinates/base.py [hunk 3]
```diff
@@ -757,8 +768,15 @@ class IObase(object):
     units = {'time': None, 'length': None, 'velocity': None}
 
     def convert_pos_from_native(self, x, inplace=True):
-        """In-place conversion of coordinate array x from native units to base units.
+        """Conversion of coordinate array x from native units to base units.
 
+        Keywords
```
Comment (kain88-de) [inline package/MDAnalysis/coordinates/base.py]: This is also not the correct numpy-style and we should add `x` in the parameter description.

File: package/MDAnalysis/coordinates/base.py [hunk 4]
```diff
@@ -777,13 +796,21 @@ def convert_pos_from_native(self, x, inplace=True):
         return x
 
     def convert_velocities_from_native(self, v, inplace=True):
-        """In-place conversion of velocities array *v* from native units to base units.
+        """Conversion of velocities array *v* from native to base units
 
+        Keywords
```
Comment (kain88-de) [inline package/MDAnalysis/coordinates/base.py]: same as above

File: package/MDAnalysis/coordinates/base.py [hunk 5]
```diff
@@ -792,13 +819,21 @@ def convert_velocities_from_native(self, v, inplace=True):
         return v
 
     def convert_forces_from_native(self, force, inplace=True):
-        """In-place conversion of forces array *force* from native units to base units.
+        """Conversion of forces array *force* from native to base units
```
Comment (kain88-de) [inline package/MDAnalysis/coordinates/base.py]: same as above

File: package/MDAnalysis/coordinates/base.py [hunk 6]
```diff
@@ -809,6 +844,13 @@ def convert_forces_from_native(self, force, inplace=True):
     def convert_time_from_native(self, t, inplace=True):
         """Convert time *t* from native units to base units.
 
+        Keywords
+        --------
```
Comment (kain88-de) [inline package/MDAnalysis/coordinates/base.py]: same as above

File: package/MDAnalysis/coordinates/base.py [hunk 7]
```diff
@@ -1082,54 +1162,70 @@ def _sliced_iter(self, start, stop, step):
...
+        except TypeError:  # if _read_frame not implemented
+            raise TypeError("{0} does not support slicing."
+                            "".format(self.__class__.__name__))
+
+    def check_slice_indices(self, start, stop, step):
+        """Check frame indices are valid and clip to fit trajectory
+
+        Parameters
+        ----------
+        start, stop, step
```
Comment (kain88-de) [inline package/MDAnalysis/coordinates/base.py]: each of these should be on it's own line and have a expected type.

File: package/MDAnalysis/coordinates/base.py [hunk 8]
```diff
@@ -1082,54 +1162,70 @@ def _sliced_iter(self, start, stop, step):
...
+          Values representing the slice indices.
+          Can use `None` to use defaults of (0, -1, and 1)
+          respectively.
 
-        Return the start stop and step indices
+        Returns
+        -------
+        start, stop step
```
Comment (kain88-de) [inline package/MDAnalysis/coordinates/base.py]: same here.

File: package/MDAnalysis/coordinates/base.py [hunk 9]
```diff
@@ -1082,54 +1162,70 @@ def _sliced_iter(self, start, stop, step):
...
         """
-        if not (((type(start) == int) or (start is None)) and
-                ((type(stop) == int) or (stop is None)) and
-                ((type(step) == int) or (step is None))):
-            raise TypeError("Slice indices are not integers")
+        for var, varname in (
+                (start, 'start'),
+                (stop, 'stop'),
+                (step, 'step')
+        ):
```
Comment (kain88-de) [inline package/MDAnalysis/coordinates/base.py]: I personally don't like that the loop starting `:` is at the same indentation as the `for` keywords. I would prefer a formating like this.

```python
 for var, varname in ((start, 'start'),
                      (stop, 'stop'),
                      (step, 'step')):
```

File: package/MDAnalysis/coordinates/base.py [hunk 10]
```diff
@@ -832,6 +875,13 @@ def convert_time_from_native(self, t, inplace=True):
     def convert_pos_to_native(self, x, inplace=True):
         """Conversion of coordinate array x from base units to native units.
 
+        Keywords
```
Comment (kain88-de) [inline package/MDAnalysis/coordinates/base.py]: Parameters

File: package/MDAnalysis/coordinates/base.py [hunk 11]
```diff
@@ -850,13 +901,21 @@ def convert_pos_to_native(self, x, inplace=True):
         return x
 
     def convert_velocities_to_native(self, v, inplace=True):
-        """In-place conversion of coordinate array *v* from base units to native units.
+        """Conversion of coordinate array *v* from base to native units
+
+        Keywords
```
Comment (kain88-de) [inline package/MDAnalysis/coordinates/base.py]: Parameters

File: package/MDAnalysis/coordinates/base.py [hunk 12]
```diff
@@ -865,13 +924,21 @@ def convert_velocities_to_native(self, v, inplace=True):
         return v
 
     def convert_forces_to_native(self, force, inplace=True):
-        """In-place conversion of force array *force* from base units to native units.
+        """Conversion of force array *force* from base to native units.
+
+        Keywords
```
Comment (kain88-de) [inline package/MDAnalysis/coordinates/base.py]: Parameters

File: package/MDAnalysis/coordinates/base.py [hunk 13]
```diff
@@ -882,6 +949,13 @@ def convert_forces_to_native(self, force, inplace=True):
     def convert_time_to_native(self, t, inplace=True):
         """Convert time *t* from base units to native units.
 
+        Keywords
```
Comment (kain88-de) [inline package/MDAnalysis/coordinates/base.py]: Parameters

**Comment by kain88-de** [comment]
I think I now have found all doc-string formatting errors. For most of them a search and replace should be fine. Otherwise this looks fine to me as well.

File: package/MDAnalysis/coordinates/base.py (hunk 7 above)
Comment (richardjgowers) [inline package/MDAnalysis/coordinates/base.py]: I think it makes sense to group arguments that share types, and it's allowed:

When two or more input parameters have exactly the same type, shape and description, they can be combined:

x1, x2 : array_like
    Input arrays, description of `x1`, `x2`.

**Pull request merged**